### ISSUE TITLE
Particles not scaling as expected

### ISSUE THREAD

**Issue by webtech1980** [squarefeet/ShaderParticleEngine#66]
I might be wrong but it seems that there is an issue with scaling. When I test with the examples things look very different when full screened compared to 1/4 screen. It seems like the particle sizes are not being scaled.

**Comment by Usnul** [comment]
```js  
 var viewportScale = this.renderer.size.y / 2.0;
        var groupViewportScale = speGroup.uniforms.scale.value;
        if (groupViewportScale !== viewportScale) {
            speGroup.uniforms.scale.value = viewportScale;
            speGroup.material.needsUpdate = true;
        }
```

ensure you have following in your vertex shader:

```glsl

uniform float scale;
....
 if( hasPerspective == 1 ) {
        pointSize = pointSize * ( scale / length( pos.xyz ) );
}
```

also, make sure scale is in uniforms array:
```js
// Create uniforms
    that.uniforms = {
        duration: {type: 'f', value: that.maxAge},
        texture: {type: 't', value: that.texture},
        hasPerspective: {type: 'i', value: that.hasPerspective},
        colorize: {type: 'i', value: that.colorize},
        scale: {type: "f", value: that.scale || 300}
    };
```

**Comment by webtech1980** [comment]
Thanks I'll give that a try

**Comment by squarefeet** [comment]
Fixed in 1.0.0. See `scale` option for `SPE.Group` [here](https://squarefeet.github.io/ShaderParticleEngine/docs/api/global.html#GroupOptions)
